lib/cocoapods/target/build_settings.rb

```diff
@@ -629,9 +629,9 @@ def initialize(target, non_library_spec = nil)
           libraries = []
           if non_library_xcconfig? || target.build_as_dynamic?
             libraries.concat vendored_static_frameworks.map { |l| File.basename(l, '.framework') }
+            libraries.concat libraries_to_import
```

Comment:
should this only be dynamic libraries ?

Reply:
I believe its for all libraries. It's only for building a dynamic library, but it needs to know all of its static and dynamic library dependencies.

Reply:
Does this mirror what we have for frameworks?

Reply:
Yes, frameworks always get added to `OTHER_LDFLAGS`.